src/libstd/keyword_docs.rs

```diff
@@ -1578,12 +1578,108 @@ mod type_keyword {}
 
 #[doc(keyword = "unsafe")]
 //
-/// Code or interfaces whose [memory safety] cannot be verified by the type system.
+/// Code or interfaces whose [memory safety] cannot be verified by the type
+/// system.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
+/// The `unsafe` keyword has two uses: to declare the existence of contracts the
+/// compiler can't check, and to declare that a programmer has checked that
+/// these contracts have been upheld.
+///
+/// Unsafe Rust add some capabilities to Safe Rust:
+///
+/// - Dereference [raw pointers]
+/// - Call `unsafe` functions (including foreign functions, compiler intrinsics,
+/// and the raw allocator)
+/// - Implement `unsafe` [`trait`]s
+/// - Mutate [`static`]s (including [`extern`]al ones)
+/// - Access fields of [`union`]s
+///
+/// # Dereferencing raw pointers
+///
+/// ```rust
+/// let a = 3;
+/// let b = &a as *const _;
+/// // SAFETY: we know that b will always be a valid address.
+/// let _ = unsafe { *b };
+/// ```
+///
+/// # Marking elements as unsafe
+///
+/// `unsafe` can be used on functions. Note that functions declared in
+/// [`extern`] blocks are implicitly marked as `unsafe`. Methods can also be
+/// declared as `unsafe`:
+///
+/// ```rust
+/// #![allow(dead_code)]
```

Comment:
IMO this distracts a little from the example.
```suggestion
/// # #![allow(dead_code)]
```

Reply:
That's purely a mistake on my end, I intended to hide it from the start